internal/controller/pkg/revision/imageback.go

```diff
@@ -74,35 +73,44 @@ func NewImageBackend(cache xpkg.Cache, fetcher xpkg.Fetcher, opts ...ImageBacken
 
 // Init initializes an ImageBackend.
 func (i *ImageBackend) Init(ctx context.Context, bo ...parser.BackendOption) (io.ReadCloser, error) {
+	// NOTE(hasheddan): we use nestedBackend here because simultaneous
+	// reconciles of providers or configurations can lead to the package
+	// revision being overwritten mid-execution in the shared image backend when
+	// it is a member of the image backend struct. We could introduce a lock
+	// here, but there is no reason why a given reconcile should require
+	// exclusive access to the image backend other than its poor design. We
+	// should consider restructuring the parser backend interface to better
+	// accommodate for shared, thread-safe backends.
```

Comment:
Can you raise an issue to track restructuring this please. I'm okay merging this as a scoped fix but I had a tough time following what was happening. It's possible I'm just suffering from residual vacation brain, but `*nestedBackend` having to pretend to be a `parser.Backend` in order to allow us to pass in a PackageRevision seems pretty unintuitive.

Reply:
@negz yes, as evidenced by the note I don’t think this is a good design, but I do think it’s a safer shorty term fix than moving away from the existing parser interface today. There are a number of performance optimizations that could also be implemented as part of a refactor 👍🏻

Reply:
@negz issue opened here https://github.com/crossplane/crossplane/issues/2803